### PR TITLE
Update Sparkmagic to use SparkSession for Spark2

### PR DESCRIPTION
**Pull request by jeffersonezra**
In Spark2, we have SparkSession (variable name spark) instead of HiveContext (variable name sqlContext). Updated SqlQuery to support this and added corresponding tests. Also bumped the patch version.

### CODE REVIEW

File: sparkmagic/sparkmagic/tests/test_sqlquery.py [hunk 1]
```diff
@@ -194,3 +194,79 @@ def test_unicode_sql():
         assert False
     except NotImplementedError:
         pass
+
+@with_setup(_setup, _teardown)
+def test_pyspark_livy_sql_options_spark2():
+    try:
+
```
Comment (aggFTW) [inline sparkmagic/sparkmagic/tests/test_sqlquery.py]: Remove extra line

File: sparkmagic/sparkmagic/tests/test_sqlquery.py [hunk 2]
```diff
@@ -194,3 +194,79 @@ def test_unicode_sql():
...
+    finally:
+        defaults = {
+            conf.use_spark_session.__name__: False,
+        }
+        conf.override_all(defaults)
+
+@with_setup(_setup, _teardown)
+def test_scala_livy_sql_options_spark2():
+    try:
+
```
Comment (aggFTW) [inline sparkmagic/sparkmagic/tests/test_sqlquery.py]: Remove empty line

File: sparkmagic/sparkmagic/tests/test_sqlquery.py [hunk 3]
```diff
@@ -194,3 +194,79 @@ def test_unicode_sql():
...
+                              .format(LONG_RANDOM_VARIABLE_NAME, query,
+                                      LONG_RANDOM_VARIABLE_NAME, conf.pyspark_sql_encoding())))
+
+        sqlquery = SQLQuery(query, samplemethod='sample', samplefraction=0.33, maxrows=3234)
+        assert_equals(sqlquery._pyspark_command(),
+                      Command(u'for {} in spark.sql(u"""{} """).toJSON().sample(False, 0.33).take(3234): '
+                              u'print({}.encode("{}"))'\
+                              .format(LONG_RANDOM_VARIABLE_NAME, query,
+                                      LONG_RANDOM_VARIABLE_NAME, conf.pyspark_sql_encoding())))
+
```
Comment (aggFTW) [inline sparkmagic/sparkmagic/tests/test_sqlquery.py]: Remove empty line

File: sparkmagic/sparkmagic/tests/test_sqlquery.py [hunk 4]
```diff
@@ -194,3 +194,79 @@ def test_unicode_sql():
...
+                      Command('spark.sql("""{}""").toJSON.collect.foreach(println)'.format(query)))
+
+        sqlquery = SQLQuery(query, samplemethod='sample', samplefraction=0.25, maxrows=-1)
+        assert_equals(sqlquery._scala_command(),
+                      Command('spark.sql("""{}""").toJSON.sample(false, 0.25).collect.foreach(println)'.format(query)))
+
+        sqlquery = SQLQuery(query, samplemethod='sample', samplefraction=0.33, maxrows=3234)
+        assert_equals(sqlquery._scala_command(),
+                      Command('spark.sql("""{}""").toJSON.sample(false, 0.33).take(3234).foreach(println)'.format(query)))
+
```
Comment (aggFTW) [inline sparkmagic/sparkmagic/tests/test_sqlquery.py]: Remove empty line

**Comment by tc0312** [comment]
I've some concerns on adding a global config for this. I would prefer a per session capability detection approach.

**Comment by aggFTW** [comment]
Is there a way to do detect that once a session is created in Livy?

**Comment by tc0312** [comment]
Yes. We can execute a statement to check does a variable exist.

**Comment by aggFTW** [comment]
When there's more than 2 options (created vs not created), for example a name change, wouldn't this increase the latency?

What about Livy returning the session version on session creation as a field? Would that be a viable approach?

**Comment by tc0312** [comment]
In practice, name change will not happen because everyone is follow spark-shell's naming. There's a pending PR to eliminate the 1 second update latency. Also, it's possible to batch submit statements.
The latency caused by Livy should be sub-second and acceptable.

We have explored the option to expose them in Livy via appInfo. It's definitely doable, I haven't seen anyone opposing the idea. However given the large amount of changes happening right now in Livy (Spark 2.0 and recovery), it will take a while before it gets implemented.

**Comment by aggFTW** [comment]
Got it. Do you want to revisit this once it's exposed via appInfo or do you foresee us running into problems with this approach?

**Pull request merged**